internal/services/iothub/iothub_resource.go

```diff
@@ -894,6 +902,12 @@ func resourceIotHubRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		}
 
 		d.Set("min_tls_version", properties.MinTLSVersion)
+
+		if properties.DisableLocalAuth != nil {
+			if err := d.Set("local_authentication_enabled", !*properties.DisableLocalAuth); err != nil {
+				return fmt.Errorf("setting `local_authentication_enabled` in IoTHub %q: %+v", id.Name, err)
+			}
+		}
```

Comment:
We should always set a value into state, since it defaults to true we should default that here as well
```suggestion
		localAuthenticationEnabled := true
		if properties.DisableLocalAuth != nil {
			localAuthenticationEnabled = !*properties.DisableLocalAuth
		}
		d.Set("local_authentication_enabled", localAuthenticationEnabled)
```